lib/modules.nix

```diff
@@ -817,28 +817,25 @@ let
       };
 
   # Merge definitions of a value of a given type.
-  mergeDefinitions = loc: type: defs: rec {
```

Comment:
The ofborg Evaluation Performance Report shows a slight increase in heap allocations, and it seems that this `rec` may have saved some.
The new `let` adds another `Env`, and it's unlikely to be released because of, say, the `isDefined` closure and the fact that references to options are often retained.